### PR TITLE
Feature mass transfer bc

### PR DESCRIPTION
**Pull request by SamueleMeschini**
## Proposed changes

Add a MassFlux boundary condition as described in https://github.com/RemDelaporteMathurin/FESTIM/issues/531

## Types of changes

What types of changes does your code introduce to FESTIM?

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Code refactoring
- [ ] Documentation Update (if none of the other choices apply)
- [ ] New tests

## Checklist

- [x] Black formatted
- [x] Unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

## Further comments

Tested on the same example in https://github.com/RemDelaporteMathurin/FESTIM/issues/531.

### CODE REVIEW

**Comment by RemDelaporteMathurin** [comment]
Hi @SamueleMeschini , thanks for this PR! 

I've been moving houses recently and will have a look at it ASAP!

File: festim/_version.py [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# coding: utf-8
+# file generated by setuptools_scm
+# don't change, don't track in version control
+__version__ = version = '0.1.dev2157+g5abf34d.d20221108'
+__version_tuple__ = version_tuple = (0, 1, 'dev2157', 'g5abf34d.d20221108')
```
Comment (RemDelaporteMathurin) [inline festim/_version.py:5]: This file should not be tracked in version control. Can you please remove it?

File: festim/boundary_conditions/fluxes/mass_flux.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+class MassFlux(FluxBC):
+    """
+    FluxBC subclass for advective mass flux
+    -D * grad(c) * n = h_mass * (c - c_ext)
+
+    Args:
+        h_mass (float or sp.Expr): mass transfer coefficient (m/s)
+        c_ext (float or sp.Expr): external concentration (1/m3)
+        surfaces (list or int): the surfaces of the BC
+    """
```
Comment (RemDelaporteMathurin) [inline festim/boundary_conditions/fluxes/mass_flux.py:15]: Any chance we could add a reference in the docstrings?

**Review by RemDelaporteMathurin — CHANGES_REQUESTED** [review]
Thanks @SamueleMeschini for this addition! I'm certain it will be used a lot!

A few things that need to be addressed:
- can you change the branch this merges in to `RemDelaporteMathurin:dev` instead of `RemDelaporteMathurin:main` ?
- Can you remove the _version.py file, it shouldn't be included in version control
- Do you think you could add a simple test that checks this class works? See [this example](https://github.com/RemDelaporteMathurin/FESTIM/blob/60a1098def78c88f332145255ed11f3af8343f88/test/unit/test_boundary_conditions.py#L502-L508)

File: festim/_version.py (hunk 1 above)
Comment (SamueleMeschini) [reply festim/_version.py]: Removed!

File: festim/boundary_conditions/fluxes/mass_flux.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+import fenics as f
+import sympy as sp
+
+
+class MassFlux(FluxBC):
+    """
+    FluxBC subclass for advective mass flux
+    -D * grad(c) * n = h_mass * (c - c_ext)
+
+    Args:
```
Comment (RemDelaporteMathurin) [inline festim/boundary_conditions/fluxes/mass_flux.py:11]:
```suggestion
    Args: ciao bella
```

File: festim/boundary_conditions/fluxes/mass_flux.py (hunk 2 above)
Comment (SamueleMeschini) [reply festim/boundary_conditions/fluxes/mass_flux.py]: done!

File: festim/boundary_conditions/fluxes/mass_flux.py [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+        self.h_coeff = h_coeff
+        self.c_ext = c_ext
+        super().__init__(surfaces=surfaces, field=0) 
+
+    def create_form(self, T, solute): 
+        h_coeff = f.Expression(sp.printing.ccode(self.h_coeff), t=0, degree=1)
+        c_ext = f.Expression(sp.printing.ccode(self.c_ext), t=0, degree=1)
+
+        self.form = -h_coeff * (solute - c_ext)
+        self.sub_expressions = [h_coeff, c_ext]
```
Comment (RemDelaporteMathurin) [inline festim/boundary_conditions/fluxes/mass_flux.py:30]:
```suggestion
        self.sub_expressions = [h_coeff, c_ext]

```

black requires blank line at the end of file

File: test/unit/test_boundary_conditions.py [hunk 5]
```diff
@@ -507,6 +507,13 @@ def test_recomb_flux():
     my_BC = festim.RecombinationFlux(surfaces=[0], Kr_0=expr, E_Kr=expr, order=2)
     my_BC.create_form(T, c)
 
+def test_mass_flux():
+    expr = 2 + festim.x
+    T = fenics.Expression(sp.printing.ccode(expr), degree=1, t=0)
+    c = fenics.Expression(sp.printing.ccode(expr), degree=1, t=0)
+
+    my_BC = festim.MassFlux(surfaces=0, h_coeff=expr, c_ext=expr)
+    my_BC.create_form(T,c)
```
Comment (RemDelaporteMathurin) [inline test/unit/test_boundary_conditions.py:516]:
```suggestion
    my_BC.create_form(T, c)
```

black requires space between functions/methods arguments

**Review by RemDelaporteMathurin — CHANGES_REQUESTED** [review]
Almost there! Just a few remarks regarding formatting.

File: test/unit/test_boundary_conditions.py [hunk 6]
```diff
@@ -507,6 +507,13 @@ def test_recomb_flux():
     my_BC = festim.RecombinationFlux(surfaces=[0], Kr_0=expr, E_Kr=expr, order=2)
     my_BC.create_form(T, c)
 
+def test_mass_flux():
```
Comment (RemDelaporteMathurin) [inline test/unit/test_boundary_conditions.py:510]:
```suggestion

def test_mass_flux():
```

black requires 2 lines between functions

**Comment by SamueleMeschini** [comment]
thank you guys for the help!

File: festim/boundary_conditions/fluxes/mass_flux.py [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+    """
+    FluxBC subclass for advective mass flux
+    -D * grad(c) * n = h_mass * (c - c_ext)
+
+    Args:
+        h_mass (float or sp.Expr): mass transfer coefficient (m/s)
+        c_ext (float or sp.Expr): external concentration (1/m3)
+        surfaces (list or int): the surfaces of the BC
+
+    Reference: Bergman, T. L., Bergman, T. L., Incropera, F. P., Dewitt, D. P.,
```
Comment (RemDelaporteMathurin) [inline festim/boundary_conditions/fluxes/mass_flux.py:16]:
```suggestion
    Reference: Bergman, T. L., Bergman, T. L., Incropera, F. P., Dewitt, D. P.,
```

File: festim/boundary_conditions/fluxes/mass_flux.py [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+        surfaces (list or int): the surfaces of the BC
+
+    Reference: Bergman, T. L., Bergman, T. L., Incropera, F. P., Dewitt, D. P., 
+    & Lavine, A. S. (2011). Fundamentals of heat and mass transfer. John Wiley & Sons.
+    """
+
+    def __init__(self, h_coeff, c_ext, surfaces) -> None:
+        self.h_coeff = h_coeff
+        self.c_ext = c_ext
+        super().__init__(surfaces=surfaces, field=0)
```
Comment (RemDelaporteMathurin) [inline festim/boundary_conditions/fluxes/mass_flux.py:23]:
```suggestion
        super().__init__(surfaces=surfaces, field=0)
```

File: festim/boundary_conditions/fluxes/mass_flux.py [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+    Reference: Bergman, T. L., Bergman, T. L., Incropera, F. P., Dewitt, D. P., 
+    & Lavine, A. S. (2011). Fundamentals of heat and mass transfer. John Wiley & Sons.
+    """
+
+    def __init__(self, h_coeff, c_ext, surfaces) -> None:
+        self.h_coeff = h_coeff
+        self.c_ext = c_ext
+        super().__init__(surfaces=surfaces, field=0) 
+
+    def create_form(self, T, solute):
```
Comment (RemDelaporteMathurin) [inline festim/boundary_conditions/fluxes/mass_flux.py:25]:
```suggestion
    def create_form(self, T, solute):
```

**Review by RemDelaporteMathurin — CHANGES_REQUESTED** [review]
A few more syntax problems, but I'll fix them myself and then merge this.

**Pull request merged**